streams/src/main/java/org/apache/kafka/streams/state/internals/MemoryLRUCache.java

```diff
@@ -105,14 +100,28 @@ public boolean persistent() {
     }
 
     @Override
-    public V get(K key) {
-        return this.map.get(key);
+    public boolean open() {
+        return open;
+    }
+
+    @Override
+    public  V get(K key) {
+        lock.readLock().lock();
+        try {
+            return this.map.get(key);
+        } finally {
+            lock.readLock().unlock();
+        }
     }
 
     @Override
     public void put(K key, V value) {
-        this.map.put(key, value);
-        this.keys.add(key);
+        lock.writeLock().lock();
```

Comment:
Yeah, I don't know if we do (depends on whether locking is a bottleneck here). And even if we did, it makes sense to do that separately instead of this PR.